This file is for exploring the case study. It is originally placed at the /root folder.

In [1]:
import numpy as np
import netCDF4
from Land.main import land
from CRS.crs_init import CRSConvertor

In [7]:
# read wind turbine locations
turbine_loc = np.loadtxt('Aborted/case/Turbines_At_Whitelee_Wind_Farm.csv', delimiter=',', encoding='utf-8', usecols=[1, 2])
conv = CRSConvertor([55.714704134580245, -4.364543821199962, 55.634359319706036, -4.183104393719847])
print(turbine_loc)

[[55.6891561 -4.3580354]
 [55.6866291 -4.3518556]
 [55.6952387 -4.3633451]
 [55.6934182 -4.3555815]
 [55.6859453 -4.3414069]
 [55.6896599 -4.3292643]
 [55.6964595 -4.3525545]
 [55.7028635 -4.3566589]
 [55.7045988 -4.351543 ]
 [55.7028327 -4.3463191]
 [55.7004356 -4.3407223]
 [55.6941866 -4.3231996]
 [55.6913359 -4.3185243]
 [55.6890699 -4.3130225]
 [55.687036  -4.3079828]
 [55.6840153 -4.3021339]
 [55.6811411 -4.2990084]
 [55.6788471 -4.2912424]
 [55.6802265 -4.284701 ]
 [55.6621995 -4.2430467]
 [55.6572329 -4.2390082]
 [55.655983  -4.23264  ]
 [55.6552007 -4.2252601]
 [55.7072954 -4.3467533]
 [55.7052446 -4.3407187]
 [55.7030703 -4.3359135]
 [55.6976605 -4.3226198]
 [55.6959594 -4.3175033]
 [55.693707  -4.3109932]
 [55.6916351 -4.3072165]
 [55.6892417 -4.3020673]
 [55.686783  -4.2961061]
 [55.6824166 -4.2928225]
 [55.6834495 -4.2855856]
 [55.6811579 -4.278979 ]
 [55.6715497 -4.2544861]
 [55.6661212 -4.2424683]
 [55.6609738 -4.235968 ]
 [55.6599369 -4.2287444]
 [55.709604  -4.3405915]


In [13]:
with netCDF4.Dataset('Land/raw/C3S-LC-L4-LCCS-Map-300m-P1Y-2022-v2.1.1.nc', 'r') as file:
    land = file.variables['lccs_class']
    lat = file.variables['lat']
    lon = file.variables['lon']

    # a list to store land cover classes occurred in current layout
    occur = []

    for i in range(turbine_loc.shape[0]):
        dist = np.abs(lat[:] - turbine_loc[i, 0])
        iy_min = dist.argmin()
        dist = np.abs(lon[:] - turbine_loc[i, 1])
        ix_min = dist.argmin()
        temp = land[0, iy_min, ix_min].data
        if temp not in occur:
            occur.append(int(temp))
        print(temp)
    
    print(occur)

180
180
180
180
70
180
180
180
180
180
180
180
180
110
180
70
110
110
70
100
70
110
110
180
180
180
210
180
180
70
70
70
70
180
180
110
100
70
70
180
180
180
130
180
180
180
70
180
180
100
70
100
180
180
130
130
180
180
180
110
100
180
180
100
70
100
100
130
130
130
130
180
180
110
70
110
110
180
100
70
70
70
100
70
70
110
180
70
70
110
180
180
70
70
70
70
110
180
110
100
70
180
100
130
100
70
70
70
100
100
110
70
70
100
70
70
70
70
100
70
70
70
70
70
70
70
70
70
70
70
70
70
70
70
100
100
70
100
70
70
70
70
70
70
110
70
70
110
70
70
70
70
70
70
70
70
70
70
70
70
70
110
100
110
70
100
100
70
70
70
70
180
70
70
70
180
180
100
130
70
70
70
100
100
100
100
70
70
110
70
70
70
70
70
100
130
100
70
70
180
180
70
110
70
70
70
100
70
70
110
110
70
70
70
70
[180, 70, 110, 100, 210, 130]


So, all the current wind turbines are installed in 6 main land cover categorises. They are:

70 Tree cover, needleleaved, evergreen, closed to open

100 Mosaic tree and shrub (>50%) / herbaceous cover (<50%) 

110 Mosaic herbaceous cover (>50%) / tree and shrub (<50%)

130 Grassland

180 Shrub or herbaceous cover, flooded, fresh/saline/brackish water

210 Water bodies. Only 1 turbine is at this location so we can ignore it.

Therefore, change the following categories from infeasible to feasible:
40, 50, 60, 70, 80, 90, 100, 140, 160, 170, 180